contracts/bid/src/map.rs

```diff
@@ -33,8 +30,8 @@ where
         &mut self,
         pk: PublicKey,
         bid_idx: usize,
-    ) -> Result<Option<u64>, S::Error> {
-        self.0.insert(pk.to_bytes(), bid_idx as u64)
+    ) -> Result<Option<u64>, CanonError> {
+        Ok(self.0.insert(pk.to_bytes(), bid_idx as u64)?)
```

Comment:
This change is not needed, you can remove the `Ok` and the `?` as it was before (the hamt's `insert` has the same return value of your `insert`).